### ISSUE TITLE
support for asset hashing

### ISSUE THREAD

**Issue by kylemac** [static-dev/spike-core#125]
since we swapped `static-dev/file-loader` for `static-dev/source-loader`, we lost the ability to [easily] hash the output paths. we overlook this at times because netlify handles it for us, but in instances where we have to host somewhere else, this is a valuable feature. 

it probably should support the same interface as file-loader: https://github.com/static-dev/file-loader#filename-template-placeholders  [source](https://github.com/static-dev/file-loader/blob/master/index.js#L11-L15)

**Comment by jescalan** [comment]
One issue we'll run into here is that when the assets are hashed, you can reference them by `require`-ing their path in any given file in order to get the reference to resolve correctly. However, we have removed parsing for `require` from our jade process, and I think it also will not work for css. So we either have to hack the jade compiler again to bring this back in, or come up with another way to ensure that even hashed asset paths can stay consistent.

**Comment by kylemac** [comment]
true. i bet we can find a nice way to handle this though. we're pulling the data/info we need for it in some form already

**Comment by jescalan** [comment]
So looking at this again, I feel like the way to take care of it would be a pair of plugins for postcss/posthtml that would grab and replace asset paths with hashes. These two libraries already parse the trees, so it shouldn't be an issue for us to isolate asset paths. As long as they both use the same hashing technique, they will generate matching hashes and all will be well.
